### ISSUE TITLE
false positive for `prefer_void_to_null` w/ `Never?`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4867]
Follow-up from https://github.com/dart-lang/sdk/issues/54820#issuecomment-1927418607

**Comment by lrhn** [comment]
It's technically correct in that `Never?` is equivalent to `Null` (according to [**NORM**](https://github.com/dart-lang/language/blob/main/resources/type-system/normalization.md#normalization), which is what most people should think of as "the same type").

The code just chose to write `Null` in a more cumbersome way than necessary.
I'd expect the same warnings for:
```dart
typedef ArgleBargle = Null;
ArgleBargle f() => null;
```

It would be more understandable if it added "The type 'Never?' is equivalent to 'Null'.". That can work for type aliases too.
